tests/mqtt/unit/iot_tests_mqtt_receive.c

```diff
@@ -775,6 +776,47 @@ TEST( MQTT_Unit_Receive, InvalidPacket )
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Tests the behavior of @ref mqtt_function_receivecallback when memory
+ * allocation fails.
+ */
+TEST( MQTT_Unit_Receive, ReceiveMallocFail )
```

Comment:
How does this test prove the "_flushPacket" is invoked?

Reply:
My question is why this test would fail with the code before this change.

Reply:
If _flushPacket were not invoked, then the second call to the ReceiveCallback would fail because a partial, invalid packet would be left in the data stream.

Reply:
Make sense.